src/field_theory/polynomial_galois_group.lean

```diff
@@ -208,6 +208,62 @@ begin
   { rwa [ne.def, mul_eq_zero, map_eq_zero, map_eq_zero, ←mul_eq_zero] }
 end
 
+lemma mul_splits_in_splitting_field_of_mul {p₁ q₁ p₂ q₂ : polynomial F}
+  (hq₁ : q₁ ≠ 0) (hq₂ : q₂ ≠ 0) (h₁ : p₁.splits (algebra_map F q₁.splitting_field))
+  (h₂ : p₂.splits (algebra_map F q₂.splitting_field)) :
+  (p₁ * p₂).splits (algebra_map F (q₁ * q₂).splitting_field) :=
+begin
+  apply splits_mul,
+  { rw ← (splitting_field.lift q₁ (splits_of_splits_of_dvd _
+      (mul_ne_zero hq₁ hq₂) (splitting_field.splits _) (dvd_mul_right q₁ q₂))).comp_algebra_map,
+    exact splits_comp_of_splits _ _ h₁, },
+  { rw ← (splitting_field.lift q₂ (splits_of_splits_of_dvd _
+      (mul_ne_zero hq₁ hq₂) (splitting_field.splits _) (dvd_mul_left q₂ q₁))).comp_algebra_map,
+    exact splits_comp_of_splits _ _ h₂, },
+end
+
+lemma splits_in_splitting_field_of_comp (hq : q.nat_degree ≠ 0) :
+  p.splits (algebra_map F (p.comp q).splitting_field) :=
+begin
+  let P : polynomial F → Prop := λ r, r.splits (algebra_map F (r.comp q).splitting_field),
+  have key1 : ∀ {r : polynomial F}, irreducible r → P r,
+  { intros r hr,
+    by_cases hr' : nat_degree r = 0,
+    { exact splits_of_nat_degree_le_one _ (le_trans (le_of_eq hr') zero_le_one) },
+    obtain ⟨x, hx⟩ := exists_root_of_splits _ (splitting_field.splits (r.comp q))
+      (λ h, hr' (or.resolve_right (mul_eq_zero.mp (eq.trans nat_degree_comp.symm
+      (nat_degree_eq_zero_iff_degree_le_zero.mpr (le_of_eq h)))) hq)),
+    rw [←aeval_def, aeval_comp] at hx,
+    have h_normal : normal F (r.comp q).splitting_field := splitting_field.normal (r.comp q),
+    exact splits_of_splits_of_dvd _ (minpoly.ne_zero (normal.is_integral h_normal _))
+      (normal.splits h_normal _) (dvd_symm_of_irreducible (minpoly.irreducible
+      (normal.is_integral h_normal _)) hr (minpoly.dvd F _ hx)) },
+  have key2 : ∀ {p₁ p₂ : polynomial F}, P p₁ → P p₂ → P (p₁ * p₂),
+  { intros p₁ p₂ hp₁ hp₂,
+    by_cases h₁ : p₁.comp q = 0,
+    { cases comp_eq_zero_iff.mp h₁ with h h,
+      { rw [h, zero_mul],
+        exact splits_zero _ },
+      { exact false.rec _ (hq (by rw [h.2, nat_degree_C])) } },
```

Comment:
Interestingly, the `induction_on_irreducible` below gives you the assumption `hp₁ : p₁ ≠ 0`, so you could shorten this, if you wanted to make it a bit less general, to:
```suggestion
  by_cases h₁ : p₁.comp q = 0,
  { obtain ⟨_, q_eq⟩ := (comp_eq_zero_iff.mp h₁).resolve_left hp₁,
    exact false.rec _ (hq (by rw [q_eq, nat_degree_C])) },
```